addons/web/static/tests/tours/user_switch_tour.js

```diff
@@ -0,0 +1,162 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+
+function assertEqual(actual, expected) {
+    if (actual !== expected) {
+        throw new Error(`Assert failed: expected: ${expected} ; got: ${actual}`);
+    }
+}
+
+function logout() {
+    return [
+        {
+            content: "check we're logged in",
+            trigger: ".o_user_menu .dropdown-toggle",
+            run: "click",
+        },
+        {
+            content: "click the Log out button",
+            trigger: ".dropdown-item[data-menu=logout]",
+            run: "click",
+        },
+    ];
+}
+
+registry.category("web_tour.tours").add("test_user_switch", {
+    test: true,
+    url: "/web",
+    steps: () => [
+        ...logout(),
+        {
+            content: "Check that we're back on the quick login page",
+            trigger: ".o_user_switch",
+            run() {
+                const users = document.querySelectorAll(".o_user_avatar");
+                assertEqual(users.length, 1);
+                assertEqual(
+                    users[0].parentElement.querySelector(".fs-5").innerText.trim(),
+                    "Marc Demo"
+                );
+                users[0].click();
+            },
+        },
+        {
+            content: "Check back button to back on the quick login page",
+            trigger: ".oe_login_form .o_user_switch_btn",
+            run: "click",
+        },
+        {
+            content: "Display the login form",
+            trigger: ".o_user_switch .fa-user-circle-o",
+            run: "click",
+        },
+        {
+            content: "fill the login",
+            trigger: "input#login",
+            run: "edit admin",
+        },
+        {
+            content: "fill the password",
+            trigger: "input#password",
+            run: "edit admin",
+        },
+        {
+            content: "click on login button",
+            trigger: 'button:contains("Log in")',
+            run: "click",
+        },
+        ...logout(),
+        {
+            content: "Choice demo",
+            trigger: ".o_user_switch",
+            run() {
+                const users = document.querySelectorAll(".o_user_avatar");
+                assertEqual(users.length, 2);
+                assertEqual(
+                    users[0].parentElement.querySelector(".fs-5").innerText.trim(),
+                    "Mitchell Admin"
+                );
+                assertEqual(
+                    users[1].parentElement.querySelector(".fs-5").innerText.trim(),
+                    "Marc Demo"
+                );
+                users[1].click();
+            },
+        },
```

Comment:
Check can be done with step no need to do all complex `run` function
e.g.:
```suggestion
        {
            content: "Check if there is Mitchell Admin in user list selection",
            trigger: ".o_user_switch .list-group-item:nth-child(1):contains('Mitchell Admin')",
        },
        {
            content: "Check if there is Marc Demo in user list selection",
            trigger: ".o_user_switch .list-group-item:nth-child(2):contains('Marc Demo')",
        },
        {
            content: "Choice demo",
            trigger: ".o_user_switch .list-group-item:contains('Marc Demo')",
            run: "click",
            },
        },
```